src/python/pants/backend/native/tasks/native_external_library_fetch.py

```diff
@@ -0,0 +1,279 @@
+                                                                 'pants-conan-remote-' + str(index_num),
+                                                                 remote_url,
+                                                                 '--insert'])
+      try:
+        stdout = subprocess.check_output(add_pants_conan_remote_cmdline.split())
+        self.context.log.debug(stdout)
+      except subprocess.CalledProcessError as e:
+        if not "already exists in remotes" in e.output:
+          raise TaskError('Error adding pants-specific conan remote: {}'.format(e.output))
+
+  def copy_package_contents_from_conan_dir(self, results_dir, conan_requirement, pkg_sha):
+    """
+    Copy the contents of the fetched pacakge into the results directory of the versioned
+    target from the conan data directory.
+
+    :param results_dir: The versioned-target results directory.
+    :param conan_requirement: The `ConanRequirement` object that produced the package sha.
+    :param pkg_sha: The sha of the local conan package corresponding to the specification.
+    """
+    src = os.path.join(os.path.join(self.workdir, '.conan'),
+                       'data',
+                       conan_requirement.directory_path,
+                       'package',
+                       pkg_sha)
+    src_lib = os.path.join(src, 'lib')
+    src_include = os.path.join(src, 'include')
+    dest_lib = os.path.join(results_dir, 'lib')
+    dest_include = os.path.join(results_dir, 'include')
+    if os.path.exists(src_lib):
+      copy_tree(src_lib, dest_lib)
+    if os.path.exists(src_include):
+      copy_tree(src_include, dest_include)
+
+  def fetch_packages(self, vt, vts_results_dir):
+    """
+    Invoke the conan pex to fetch conan packages specified by a
+    `ExternalLibLibrary` target.
+    """
+    task_product = {}
+    task_product['lib_names'] = []
+
+    # NB: CONAN_USER_HOME specifies the directory to use for the .conan data directory.
+    # This will initially live under the workdir to provie easy debugging on the initial
+    # iteration of this system (a 'clean-all' will nuke the conan dir). In the future,
+    # it would be good to migrate this under ~/.cache/pants/conan for persistence.
+    with environment_as(CONAN_USER_HOME=self.workdir):
+      for pkg_spec in vt.target.packages:
+
+        conan_requirement = ConanRequirement.parse(pkg_spec)
+
+        # Prepare conan command line and ensure remote is configured properly.
+        self.ensure_conan_remote_configuration(self._conan_binary)
+        args = conan_requirement.fetch_cmdline_args
+        cmdline = self._conan_binary.pex.cmdline(args)
+        self.context.log.debug('Running conan.pex cmdline: {}'.format(cmdline))
+
+        # Invoke conan to pull package from remote.
+        try:
+          process = subprocess.Popen(
+            cmdline.split(),
+            cwd=vts_results_dir,
+            stdout=subprocess.PIPE
+          )
+        except OSError as e:
```

Comment:
Any particular reason `subprocess.Popen` is used here instead of `subprocess.check_output` like other places?